src/routes/safe/components/Transactions/TxsTable/ExpandedTx/TxDescription/index.jsx

```diff
@@ -91,28 +100,59 @@ const SettingsDescription = ({ removedOwner, addedOwner, newThreshold }: Descrip
 
 const CustomDescription = ({
   data, value = 0, recipient, classes,
-}: CustomDescProps) => (
-  <>
-    <Block data-testid={TRANSACTIONS_DESC_CUSTOM_VALUE_TEST_ID}>
-      <Bold>
-        Send
-        {' '}
-        {value}
-        {' '}
-        ETH
-        {' '}
-        to:
-      </Bold>
-      <EtherscanLink type="address" value={recipient} />
-    </Block>
-    <Block className={classes.txData} data-testid={TRANSACTIONS_DESC_CUSTOM_DATA_TEST_ID}>
-      <Bold>Data (hex encoded):</Bold>
-      <Paragraph size="md" noMargin>
-        {data}
-      </Paragraph>
-    </Block>
-  </>
-)
+}: CustomDescProps) => {
+  const [showTxData, setShowTxData] = useState(false)
+  return (
+    <>
+      <Block data-testid={TRANSACTIONS_DESC_CUSTOM_VALUE_TEST_ID}>
+        <Bold>
+          Send
+          {' '}
+          {value}
+          {' '}
+          ETH
+          {' '}
+          to:
+        </Bold>
+        <EtherscanLink type="address" value={recipient} />
+      </Block>
+      <Block className={classes.txData} data-testid={TRANSACTIONS_DESC_CUSTOM_DATA_TEST_ID}>
+        <Bold>Data (hex encoded):</Bold>
+        <Paragraph size="md" noMargin className={classes.txDataParagraph}>
+          {showTxData ? (
+            <>
+              {data}
+              {' '}
+              <LinkWithRef
+                aria-label="Hide details of the transaction"
+                onClick={() => setShowTxData(false)}
+                rel="noopener noreferrer"
+                target="_blank"
+                className={classes.linkTxData}
+              >
+                Show Less
+              </LinkWithRef>
+            </>
+          ) : (
+            <>
+              {shortVersionOf(data, 20)}
+              {' '}
+              <LinkWithRef
```

Comment:
I think semantically this is a button, not a link. We have `ButtonLink` component just for that

Reply:
> The HTML <a> element (or anchor element), with its href attribute, creates a hyperlink to web pages, files, email addresses, locations in the same page, or anything else a URL can address

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a